src/app/hooks/useClickTracker/index.test.jsx

```diff
@@ -0,0 +1,322 @@
+
+    expect(urlToObject(viewEventUrl)).toEqual({
+      origin: 'https://logws1363.ati-host.net',
+      pathname: '/',
+      searchParams: {
+        atc:
+          'PUB-[cps_wsoj]-[brand]-[]-[CHD=promo::2]-[news::pidgin.news.story.51745682.page]-[]-[]-[http://bbc.com/pidgin/tori-51745682]',
+        hl: expect.stringMatching(/^.+?x.+?x.+?$/),
+        idclient: expect.stringMatching(/^.+?-.+?-.+?-.+?$/),
+        lng: 'en-US',
+        p: 'news::pidgin.news.story.51745682.page',
+        r: '0x0x24x24',
+        re: '1024x768',
+        s: '598343',
+        s2: '70',
+        type: 'AT',
+      },
+    });
+  });
+
+  it('should send tracking request on click of child element (button)', async () => {
+    const { getByText } = render(
+      <WithContexts>
+        <TestComponentContainer hookProps={defaultProps} />
+      </WithContexts>,
+    );
+
+    act(() => userEvent.click(getByText('Button')));
+
+    const [[viewEventUrl]] = global.fetch.mock.calls;
+
+    expect(urlToObject(viewEventUrl)).toEqual({
+      origin: 'https://logws1363.ati-host.net',
+      pathname: '/',
+      searchParams: {
+        atc:
+          'PUB-[cps_wsoj]-[brand]-[]-[CHD=promo::2]-[news::pidgin.news.story.51745682.page]-[]-[]-[http://bbc.com/pidgin/tori-51745682]',
+        hl: expect.stringMatching(/^.+?x.+?x.+?$/),
+        idclient: expect.stringMatching(/^.+?-.+?-.+?-.+?$/),
+        lng: 'en-US',
+        p: 'news::pidgin.news.story.51745682.page',
+        r: '0x0x24x24',
+        re: '1024x768',
+        s: '598343',
+        s2: '70',
+        type: 'AT',
+      },
+    });
+
+    jest.restoreAllMocks();
+  });
+
+  it('should only track clicks on the child component if clicks are tracked on both a parent and child', async () => {
+    const { getByText } = render(
+      <WithContexts>
+        <TestComponentContainerContainer />
+      </WithContexts>,
+    );
+
+    act(() => userEvent.click(getByText('Button')));
+
+    expect(global.fetch).toHaveBeenCalledTimes(1);
+
+    const [[viewEventUrl]] = global.fetch.mock.calls;
+
+    expect(urlToObject(viewEventUrl)).toEqual({
+      origin: 'https://logws1363.ati-host.net',
+      pathname: '/',
+      searchParams: {
+        atc:
+          'PUB-[cps_wsoj]-[brand]-[]-[CHD=promo::2]-[news::pidgin.news.story.51745682.page]-[]-[]-[http://bbc.com/pidgin/tori-51745682]',
+        hl: expect.stringMatching(/^.+?x.+?x.+?$/),
+        idclient: expect.stringMatching(/^.+?-.+?-.+?-.+?$/),
+        lng: 'en-US',
+        p: 'news::pidgin.news.story.51745682.page',
+        r: '0x0x24x24',
+        re: '1024x768',
+        s: '598343',
+        s2: '70',
+        type: 'AT',
+      },
+    });
+
+    jest.restoreAllMocks();
+  });
+
+  it('should allow the user to navigate after clicking on a tracked link even if the tracking request fails', done => {
+    const url = 'https://bbc.com/pidgin';
+
+    const spyFetch = jest.spyOn(global, 'fetch').mockImplementation(() => {
```

Comment:
global fetch is already mocked

so you can just mock it again 

```js
    global.fetch = jest.fn(() => {
      throw new Error('Failed to fetch');
    });
```

`jest.fn(() => {` is a another way to write `mockImplementation` here